android/java/org/chromium/chrome/browser/settings/BraveSyncScreensPreference.java

```diff
@@ -1363,32 +1130,65 @@ private void setAddMobileDeviceLayout() {
       getActivity().runOnUiThread(new Runnable() {
           @Override
           public void run() {
-              // TODO(sergz): Uncomment sync service impl when we fully migrate on sync v2
-              // if (null != mSyncService) {
-              //     String seed = BravePrefServiceBridge.getInstance().getSyncSeed();
-              //     if (null == seed || seed.isEmpty()) {
-              //         startTimeoutTimerWithPopup(getResources().getString(R.string.brave_sync_loading_data_title));
-              //         // Init to receive new seed
-              //         mSyncService.onSetupSyncNewToSync(mDeviceName);
-              //     }
-              // }
-
-              BraveActivity mainActivity = BraveRewardsHelper.getBraveActivity();
+              BraveActivity mainActivity = BraveActivity.getBraveActivity();
               if (null != mainActivity && null != mainActivity.mBraveSyncWorker) {
-                  SharedPreferences sharedPref = getActivity().getApplicationContext().getSharedPreferences(BraveSyncWorker.PREF_NAME, 0);
-                  String seed = sharedPref.getString(BraveSyncWorker.PREF_SEED, null);
-                  if (null == seed || seed.isEmpty()) {
-                      startTimeoutTimerWithPopup(getResources().getString(R.string.brave_sync_loading_data_title));
-                      // Init to receive new seed
-                      mainActivity.mBraveSyncWorker.InitSync(true, true);
+                  String seedHex =
+                          mainActivity.mBraveSyncWorker.GetSeedHexFromWords(GetCodephrase());
+                  if (null == seedHex || seedHex.isEmpty()) {
+                      // Give up, seed must be valid
+                      Log.e(TAG, "setAddMobileDeviceLayout seedHex is empty");
+                      assert false;
                   } else {
-                      mSyncScreensObserver.onSeedReceived(seed, false, true);
+                      fillQrCode(seedHex);
```

Comment:
What is the reason of converting sync words to hex and generate QR code?
We will have to covert it back to sync words again to save when scanning and because of this desktop has to do the conversion when generating QR code

Reply:
Is it because qr code generator will complain the input too long? Because the code generator of chromium I use will fail to generate the code because of it.

Reply:
The maximum input bytes is 84, so the passphrase will obviously exceed that. Our seed is 32 bytes and hex encode will make it 64 which is still in the range.